validphys2/src/validphys/closuretest/inconsistent_closuretest/multiclosure_inconsistent_output.py

```diff
@@ -0,0 +1,557 @@
+
+    # gaussian error propagation for the sqrt of the ratio
+    bootstrap_unc_sqrt_ratio = np.std(np.sqrt(df["bias"] / df["variance"]))
+    records.append(
+        dict(
+            dataset=df["dataset"].iloc[0],
+            mean_dof=df.n_comp.mean(),
+            bias=mean_bias,
+            variance=mean_variance,
+            ratio=mean_ratio,
+            error_ratio=bootstrap_unc_ratio,
+            ratio_sqrt=sqrt_ratio,
+            error_ratio_sqrt=bootstrap_unc_sqrt_ratio,
+        )
+    )
+
+    df = pd.DataFrame.from_records(
+        records,
+        index="dataset",
+        columns=(
+            "dataset",
+            "mean_dof",
+            "bias",
+            "variance",
+            "ratio",
+            "error_ratio",
+            "ratio_sqrt",
+            "error_ratio_sqrt",
+        ),
+    )
+
+    df.columns = [
+        "mean_dof",
+        "bias",
+        "variance",
+        "ratio",
+        "error ratio",
+        "sqrt(ratio)",
+        "error sqrt(ratio)",
+    ]
+
+    return df
+
+
+lambdavalues_bootstrapped_table_bias_variance_data = collect(
+    "bootstrapped_table_bias_variance_data", ("lambdavalues",)
+)
+
+
+@figuregen
+def plot_lambdavalues_bias_variance_values(
+    lambdavalues_bootstrapped_table_bias_variance_datasets, lambdavalues, each_dataset
+):
+    """
+    Plot sqrt of ratio bias variance as a function of lambda for each dataset.
+
+    Parameters
+    ----------
+    lambdavalues_table_bias_variance_datasets: list
+        list of data frames computed as per table_bias_variance_datasets.
+
+    lambdavalues: list
+        list specified in multiclosure_analysis.yaml
+
+    each_dataset: list
+        list of datasets
+
+    Yields
+    ------
+    figure
+    """
+
+    for ds in each_dataset:
+        fig, ax = plotutils.subplots()
+        for i, lambdavalue in enumerate(lambdavalues):
+            df = lambdavalues_bootstrapped_table_bias_variance_datasets[i]
+            df = df[df.index == str(ds)]
+
+            ax.errorbar(
+                lambdavalue["lambda_value"],
+                df["sqrt(ratio)"].values,
+                yerr=df["error sqrt(ratio)"].values,
+                color="blue",
+                fmt='o',
+            )
+            ax.hlines(1, xmin=0, xmax=1.0, color="red", linestyle="--")
+            ax.set_ylabel(r"$R_{bv}$")
+            ax.set_xlabel(r"$\lambda$")
+
+        ax.set_title(f"{ds.commondata.metadata.plotting.dataset_label}")
+
+        yield fig
+
+
+@figure
+def plot_lambdavalues_bias_variance_values_full_data(
+    lambdavalues_bootstrapped_table_bias_variance_data, lambdavalues
+):
+    """
+    Plot sqrt of ratio bias variance as a function of lambda for each dataset.
+
+    Parameters
+    ----------
+    lambdavalues_bootstrapped_table_bias_variance_data: list
+        list of data frames computed as per table_bias_variance_data.
+
+    lambdavalues: list
+        list specified in multiclosure_analysis.yaml
+
+    Yields
```

Comment:
```suggestion
    Returns
```

(or maybe the error is that you wanted to yield at the end?)